pandas/tseries/tests/test_timezones.py

```diff
@@ -533,6 +533,21 @@ def test_ambiguous_nat(self):
         di_test = DatetimeIndex(times, tz='US/Eastern')
         self.assert_numpy_array_equal(di_test, localized)
 
+    def test_nonexistent_raise_coerce(self):
+        from pytz.exceptions import NonExistentTimeError
+        times = ['2015-03-08 01:00', '2015-03-08 02:00', '2015-03-08 03:00']
+        index = DatetimeIndex(times)
+        tz = 'US/Eastern'
+        self.assertRaises(NonExistentTimeError,
+                          index.tz_localize, tz=tz)
+        self.assertRaises(NonExistentTimeError,
+                          index.tz_localize, tz=tz, errors='raise')
+        localized = index.tz_localize(tz=tz, errors='coerce')
+        test_times = ['2015-03-08 01:00-05:00', 'NaT',
+                      '2015-03-08 03:00-04:00']
+        test_localized = DatetimeIndex(test_times)
+        self.assert_numpy_array_equal(localized, test_localized)
```

Comment:
I can not compare their equality using ``tm.assert_index_equal``, because ``expected`` has dtype ``datetime64[ns, US/Eastern]``, and ``result`` is a plain ``datetime64[ns]``.